OpenSim/Common/Component.h

```diff
@@ -2201,37 +2201,23 @@ OpenSim_DECLARE_ABSTRACT_OBJECT(Component, Object);
     // End of System Creation and Access Methods.
     //@} 
 
-#if defined(__clang__)
-    #pragma clang diagnostic push
-    #pragma clang diagnostic ignored "-Wunsupported-friend"
-#endif
-    template<class C>
-    friend void Socket<C>::finalizeConnection(const Component& root);
-    template<class T>
-    friend void Input<T>::finalizeConnection(const Component& root);
-#if defined(__clang__)
-    #pragma clang diagnostic pop
-#endif
+public:
 
-    /** Utility method to find a component in the list of sub components of
-    this component and any of their sub components, etc..., by name or state
-    variable name. The search can be sped up considerably if the "path" or even
-    partial path name is known. For example name = "forearm/elbow/elbow_flexion"
-    will find the Coordinate component of the elbow joint that connects the
-    forearm body in linear time (linear search for name at each component level.
-    Whereas supplying "elbow_flexion" requires a tree search. Returns NULL if
-    Component of that specified name cannot be found. If the name provided is a
-    component's state variable name and a StateVariable pointer is provided, the
-    pointer will be set to the StateVariable object that was found. This
-    facilitates the getting and setting of StateVariables by name. 
+    /** Find a component in the list of sub components of
+    this component and any of their sub components, etc..., by name or path
```

Comment:
Wording is a bit confusing. Perhaps "Find a Component to which this Component is an ancestor---in other words, a Component that is directly owned by this Component (i.e., appears in one of its lists of sub-components) or is owned by one of its sub-components, sub-sub-components, etc. The Component can be found by type (by specifying a template argument) and either path or name."